FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/tls_freertos.c

```diff
@@ -0,0 +1,604 @@
+    mbedtlsError = mbedtls_entropy_add_source( &entropyContext,
+                                               mbedtls_platform_entropy_poll,
+                                               NULL,
+                                               32,
+                                               MBEDTLS_ENTROPY_SOURCE_STRONG );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to add entropy source, with mbedTLS high level error %s"
+                    " and with low level error %s.",
+                    mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                    mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+        returnStatus = TLS_TRANSPORT_API_ERROR;
+    }
+
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        /* Seed the random number generator. */
+        mbedtlsError = mbedtls_ctr_drbg_seed( &ctrDrgbContext,
+                                              mbedtls_entropy_func,
+                                              &entropyContext,
+                                              NULL,
+                                              0 );
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to seed PRNG, with mbedTLS high level error %s"
+                        " and with low level error %s.",
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = TLS_TRANSPORT_API_ERROR;
+        }
+    }
+
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        LogDebug( ( "Successfully Initialized mbedTLS." ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+TlsTransportStatus_t TLS_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
+                                           const char * pHostName,
+                                           uint16_t port,
+                                           const NetworkCredentials_t * pNetworkCredentials,
+                                           uint32_t receiveTimeoutMs,
+                                           uint32_t sendTimeoutMs )
+{
+    TlsTransportStatus_t returnStatus = TLS_TRANSPORT_SUCCESS;
+    BaseType_t socketStatus = 0;
+
+    if( ( pNetworkContext == NULL ) ||
+        ( pHostName == NULL ) ||
+        ( pNetworkCredentials == NULL ) )
+    {
+        LogError( ( "Arguments cannot be NULL. pNetworkContext=%p, "
+                    "pHostName=%p, pNetworkCredentials=%p.",
+                    pNetworkContext,
+                    pHostName,
+                    pNetworkCredentials ) );
+        returnStatus = TLS_TRANSPORT_INVALID_PARAMETER;
+    }
+    else if( ( pNetworkCredentials->pRootCa == NULL ) )
+    {
+        LogError( ( "pRootCa cannot be NULL." ) );
+        returnStatus = TLS_TRANSPORT_INVALID_PARAMETER;
+    }
+
+    /* Create a FreeRTOS+TCP socket. */
```

Comment:
Is this comment valid? Does this function require that the tcp socket is already created before passing the network context to it?

Reply:
As this function performs the TCP connection, should it also make the call to `FreeRTOS_socket` to create the socket?

Reply:
Nevermind, I realize that it is using the wrapper function in `sockets_freertos_wrapper.h`. I think the comment should mention
```suggestion
    /* Establish a TCP connection with the server. */
```